SSA/Projects/InstCombine/update_alive_statements.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+# Released under Apache 2.0 license as described in the file LICENSE.
+
+import re
+
+def getProofs(lines):
+    proofs = []
+    
+    proof = []
+    for line in lines:
+        if line.startswith("-- Name"):
+            proofs.append(proof)
+            proof = []
+        proof.append(line)
+    proofs.append(proof)
+
+    return proofs[0], proofs[1:]
+
+def getStatement(preamble, proof):
+    f = open("AliveTest.lean", "w")
+
+    f.write("".join(preamble))
+    rewritten = []
+    name = ""
+    for line in proof:
+        if line.startswith("  apply"):
+            rewritten.append("  print_goal_as_error")
+            name = line.split(" ")[3][0:-1]
+        else:
+            rewritten.append(line)
+    f.write("".join(rewritten))
+
+    f.close()
+    
+    import subprocess
+    x = subprocess.run("(cd ../../../; lake build SSA.Projects.InstCombine.AliveTest)",
+        shell=True, capture_output=True)
+    
+    name = "\n\ntheorem " + name + " :\n"
+
+    if x.returncode == 0:
+        return ""
+    error = x.stderr.decode("utf-8")
+    error = error.split("\n")
+    stmt = name
+    stmt += re.sub(".*error: ", "", error[2])
+    if len(error) >= 6:
+        stmt += "\n" + error[3]
+    if len(error) >= 7:
+        stmt += "\n" + error[4]
+    if len(error) >= 8:
+        stmt += "\n" + error[5]
+    if len(error) >= 9:
+        stmt += "\n" + error[6]
+    if len(error) >= 10:
+        stmt += "\n" + error[7]
+    if len(error) >= 11:
+        stmt += "\n" + error[8]
```

Comment:
This error checking code based on length looks totally crazy at first glance. I assume the length indicates something, can I have a descriptive constant name, and better yet, an explanation of what's going on here?

Reply:
Fixed.